gunrock/app/sm/sm_app.cu

```diff
@@ -60,11 +62,12 @@ double sm(
     const int           *query_row_offsets,
     const int           *query_col_indices,
     const int            num_runs,
+       string            device,
```

Comment:
Is there a reason to use a string here rather than creating an enum that's exposed through the API?

string is an immutable object, copying it down through the call stack will be much more expensive than an enum (which internally is an integer).

Reply:
@Laurawly we can use the location enum that already exists under `util::Location`, you can name it whatever, but we don't need string here.

Reply:
Sure, I can change accordingly.